secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsReceiver.java

```diff
@@ -0,0 +1,71 @@
+package de.adorsys.android.securemobilepush.sms;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.content.LocalBroadcastManager;
+import android.telephony.SmsMessage;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import de.adorsys.android.securemobilepush.BuildConfig;
+
+public class SmsReceiver extends BroadcastReceiver {
+    public static final String INTENT_FILTER_SMS = "intentFilterSms";
+    public static final String KEY_SMS_SENDER = "smsSender";
+    public static final String KEY_SMS_MESSAGE = "smsMessage";
+
+    private static final String INTENT_SMS_ACTION = "android.provider.Telephony.SMS_RECEIVED";
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (intent.getAction().equals(INTENT_SMS_ACTION)) {
+            List smsSenderNumbers =
+                    new ArrayList<>(Arrays.asList(BuildConfig.SMS_SENDER_NUMBER));
+            Bundle bundle = intent.getExtras();
+            SmsMessage[] smsMessages;
+            String messageFrom;
+            if (bundle != null) {
+                try {
+                    Object[] pdus = (Object[]) bundle.get("pdus");
+                    if (pdus != null) {
+                        smsMessages = new SmsMessage[pdus.length];
+                        for (int i = 0; i < smsMessages.length; i++) {
+                            String format = bundle.getString("format");
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                                smsMessages[i] = SmsMessage.createFromPdu((byte[]) pdus[i], format);
+                            } else {
+                                smsMessages[i] = SmsMessage.createFromPdu((byte[]) pdus[i]);
+                            }
+                            messageFrom = smsMessages[i].getOriginatingAddress();
+                            if (smsSenderNumbers.contains(messageFrom)) {
+                                String messageBody = getSmsCode(smsMessages[i].getMessageBody());
```

Comment:
Fixed in commit: https://github.com/adorsys/secure-key-storage/commit/bc533c9ebe202482e9109dfc6c33f3afd226fe53